### PR TITLE
Update get-cloud-api-spec.js

### PR DESCRIPTION
**Pull request by JakeSCahill**
## Description

Makes the script more generic/extensible for future use.

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

File: scripts/fetch-from-github/fetch.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  `);
+}
+
+if (args.includes('--help') || args.includes('-h')) {
+  printHelp();
+  process.exit(0);
+}
+
+if (args.length < 4) {
+  console.error('Error: Missing arguments.');
```
Comment (Deflaimun) [inline scripts/fetch-from-github/fetch.js:48]: Error could possibly include more information to help the user.

File: scripts/fetch-from-github/fetch.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
+const fs = require('fs');
+const path = require('path');
+
+// Function to load Octokit once
+let octokitInstance = null;
+
+async function loadOctokit() {
+  if (!octokitInstance) {
+    const { Octokit } = await import('@octokit/rest');
+    octokitInstance = process.env.VBOT_GITHUB_API_TOKEN
```
Comment (Deflaimun) [inline scripts/fetch-from-github/fetch.js:10]: Should this be REDPANDA_GITHUB_TOKEN to match other usages in docs-extensions-and-macros?

File: scripts/fetch-from-github/fetch.js (hunk 1 above)
Comment (Deflaimun) [inline scripts/fetch-from-github/fetch.js]: not sure if mixing two different log levels in the same output is a good idea. Maybe wrap this in another function and call something like 

```
function logMessage(message, level = 'log') {
  if (level === 'error') {
    console.error(message);
  } else {
    console.log(message);
  }
}

// Print help message function
function printHelp() {
  logMessage(`...`) 
  
  }
...

if (args.length < 4) {
  logMessage('Error: Missing arguments.', 'error'); 
  printHelp();
  process.exit(1);
}
```

**Review by Deflaimun — APPROVED** [review]
Check minor comment

File: scripts/fetch-from-github/fetch.js (hunk 1 above)
Comment (JakeSCahill) [reply scripts/fetch-from-github/fetch.js]: thanks 👍

File: scripts/fetch-from-github/fetch.js (hunk 2 above)
Comment (JakeSCahill) [reply scripts/fetch-from-github/fetch.js:10]: this doesn't get executed in the same environment so the name isn't so important. We're just getting it from AWS, which also has a different name 😄

**Pull request merged**